### DISCUSSION TITLE
Canvas is not found

### DISCUSSION THREAD

**Discussion by HexSleeves** [gfx-rs/wgpu#3737] [Q&A]
I keep getting `Canvas is not found` when building for wasm, and I cannot figure out why? Has anyone dealt with this issue?

**Comment by austintheriot** [comment]
I was getting the same error--the code that was panicking was the following:

```rs
        let surface = instance.create_surface(window).unwrap();
```

The answer for me ended up being the to initialize the window with the canvas element I want to render to before building the window:

```rs
    let window = WindowBuilder::new()
        .with_canvas(Some(html_canvas_element))
        .build(&event_loop)
        .unwrap();
```
